app/models/merchant.rb

```diff
@@ -11,4 +11,52 @@ def self.build_from_github(auth_hash)
     #user has not been saved, will be saved outside of this method
     return merchant
   end
+
+
+  def total_revenue
+    merchant_products = self.products 
+    total_revenue = 0.0
+
+    merchant_products.each do |product|
+      product.order_items.each do |item|
+        if item.order.status == "cancelled"
+          next
+        end
+        
+        total_revenue += item.quantity * item.product.price
+      end 
+    end
+
+    return total_revenue
+  end
+
+  def revenue_for(status)
+    merchant_products = self.products 
+    total_revenue = 0.0
+
+    merchant_products.each do |product|
+      product.order_items.each do |item|
+        if item.order.status == status
+          total_revenue += item.quantity * item.product.price
+        end
+      end
+    end
+
+    return total_revenue
+  end
```

Comment:
This is great! Do you think we can use an optional parameter to combine these methods into one? (Note: I haven't tested this suggestion!)

```suggestion
  def total_revenue(status = nil)
    merchant_products = self.products 
    total_revenue = 0.0

    merchant_products.each do |product|
      product.order_items.each do |item|
        if status
           total_revenue += item.quantity * item.product.price if item.order.status == status
        else
           total_revenue += item.quantity * item.product.price unless item.order.status == "cancelled"
        end
      end 
    end

    return total_revenue
  end
```